src/wdlci/auth/refresh_token_auth.py

```diff
@@ -27,6 +28,20 @@ def access_token(self):
 
         return self._access_token
 
+    def check_refresh_token_expiry(self):
+        decoded = jwt.decode(self.refresh_token, options={"verify_signature": False})
+        expiry = datetime.datetime.fromtimestamp(decoded["exp"])
+        now = datetime.datetime.now()
+
+        if expiry < now:
+            print(f"The refresh token is expired as of {expiry}")
+            raise WdlTestCliExitException(
+                "The refresh token is expired, the GitHub actions secrets will need to be updated with a new refresh token.",
+                1,
+            )
+        else:
```

Comment:
Let's have this check if you're within 2 weeks of expiry and emit a warning message if so prompting the user to update the token. If the token isn't within this window or expired, no messages needed.